cs/codec-template.cs.j2

```diff
@@ -0,0 +1,264 @@
+        {%- if param.nullable  -%}
+            CodecUtil.EncodeNullable(clientMessage, {{ param.name }}, {{ lang_name(param.type) }}Codec.Encode)
+        {%- else -%}
+            {{ lang_name(param.type) }}Codec.Encode(clientMessage, {{ param.name }})
+        {%- endif %}
+    {% endif %}
+{%- endmacro %}
+{% macro decode_var_sized(param) -%}
+    {%- if is_var_sized_list(param.type) or is_var_sized_list_contains_nullable(param.type) -%}
+        ListMultiFrameCodec.Decode{% if is_var_sized_list_contains_nullable(param.type) %}ContainsNullable{% endif %}{% if param.nullable  %}Nullable{% endif %}(ref iterator, {{ item_type(lang_name, param.type) }}Codec.Decode)
+    {%- elif is_var_sized_map(param.type) -%}
+        MapCodec.Decode{% if param.nullable  %}Nullable{% endif %}(ref iterator, {{ key_type(lang_name, param.type) }}Codec.Decode, {{ value_type(lang_name, param.type) }}Codec.Decode)
+    {%- else -%}
+        {%- if param.nullable  -%}
+            CodecUtil.DecodeNullable(ref iterator, {{ lang_name(param.type) }}Codec.Decode)
+        {%- else -%}
+            {{ lang_name(param.type) }}Codec.Decode(ref iterator)
+        {%- endif -%}
+    {%- endif -%}
+{%- endmacro %}
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Collections;
+using System.Collections.Generic;
+using Hazelcast.Client.Protocol;
+using Hazelcast.Client.Protocol.Codec.BuiltIn;
+using Hazelcast.Client.Protocol.Util;
+using Hazelcast.IO;
+using static Hazelcast.Client.Protocol.Codec.BuiltIn.FixedSizeTypesCodec;
+using static Hazelcast.Client.Protocol.ClientMessage;
+using static Hazelcast.IO.Bits;
+{% if method.events|length != 0 %}
+using Hazelcast.Logging;
+{% endif %}
+
+{% if namespace %}
+namespace {{ namespace }}
+{% else %}
+namespace Hazelcast.Client.Protocol.Codec
+{% endif %}
+{
+    /*
+    * This file is auto-generated by the Hazelcast Client Protocol Code Generator.
+    * To change this file, edit the templates or the protocol
+    * definitions on the https://github.com/hazelcast/hazelcast-client-protocol
+    * and regenerate it.
+    */
+
+    /// <summary>
+    {% for line in method.doc.splitlines() %}
+    /// {{ line }}
+    {% endfor %}
+    ///</summary>
+    internal static class {{ service_name|capital }}{{ method.name|capital }}Codec 
+    {
+        public const int RequestMessageType = {{ '0x%06X'|format(method.request.id) }};
```

Comment:
```    
//hex: {{ '0x%06X'|format(method.request.id) }}
public const int RequestMessageType = {{ method.request.id }};
```
using the hex as a comment and keep the value as decimal 
from Java codec. This was what we discussed to be best for debug.
You may consider if it apply to .net vs etc debuging cases too.

Reply:
Will fix